src/model/formats/crypto-utils.ts

```diff
@@ -0,0 +1,38 @@
+enum KeyUsage {
+  ENCRYPT = 'encrypt',
+  DECRYPT = 'decrypt',
+}
+
+export class CryptoUtils {
+  public static readonly CIPHER_TYPE = 'AES-CBC';
+  public static readonly CIPHER_BLOCK_SIZE = 16;
+
+  public static readonly ENCODER = new TextEncoder();
+  public static readonly DECODER = new TextDecoder();
+
+  public static async getKey(
+    cipherKey: ArrayBuffer,
+    cipherIv: ArrayBuffer,
+    keyUsage: KeyUsage,
+  ): Promise<[AesCbcParams, CryptoKey]> {
+    const cryptoKey = window.crypto.subtle.importKey('raw', cipherKey, { name: CryptoUtils.CIPHER_TYPE }, false, [
+      keyUsage,
+    ]);
+    const algorithm = { name: CryptoUtils.CIPHER_TYPE, iv: cipherIv };
+    return [algorithm, await cryptoKey];
+  }
+
+  public static async decrypt(cipherKey: ArrayBuffer, cipherIv: ArrayBuffer, data: ArrayBuffer): Promise<Uint8Array> {
+    const [algorithm, key] = await CryptoUtils.getKey(cipherKey, cipherIv, KeyUsage.DECRYPT);
+    return new Uint8Array(await window.crypto.subtle.decrypt(algorithm, key, data));
+  }
+
+  public static async encrypt(cipherKey: ArrayBuffer, cipherIv: ArrayBuffer, data: string): Promise<ArrayBuffer> {
```

Comment:
For API symmetry, you probably want "data" to be of the same type for encrypt and decrypt.
Since the file contents being textual is format-specific, my suggestion is to move ENCODER and DECODER back and just take an ArrayBuffer or similar here.

Reply:
You are right! I pushed a fixup.

I haven't gotten around to re-encrypting things yet, so it didn't occur to me. But I would like to keep the encoder & decoder here to be able to share them between classes please.

Reply:
Keeping it here sounds fine. Optional suggestion: encryptText() / decryptText() functions that use the shared decoders.

Reply:
That actually makes sense, thanks! Pushed another fixup, how about this?

Reply:
Close :) added a commit with what I meant (untested, need to change the other file)

Reply:
Ah, I see, thank you! Yes, if you like it more, it works for both ForeFlight and possibly other formats. I pushed a fixup to fix ForeFlight.